irods_specific_implementation/irods_rest_logical_path_api_implementation.h

```diff
@@ -115,10 +119,371 @@ namespace irods::rest
+                        }
+                        else if ( status == CAT_NO_ROWS_FOUND) {
+                            return make_error_response(status, "Could not find the logical path you entered. You may have passed an empty string.");
+                        }
+                        else if ( status < 0 ) {
+                            return make_error_response(status, "Client experienced an error processing your trim request. Please check the logs or notify your administrator.");
+                        }
+                    }
+                }
+              } else {
+	    	if ( fscli::is_data_object( fscli::status( *conn(), decode_url(trim_input.objPath) ) ) ) {
+		    status = rcDataObjTrim(conn(), &trim_input);
+		} else {
+		    return make_error_response(USER_INCOMPATIBLE_PARAMS, "Target is not a data object.");
+		}
+              }
+              if ( status == SYS_NOT_ALLOWED ) {
+                return  make_error_response(status, "You have tried to trim a data object or collection you don't have permissions for.");
+              }
+              else if ( status == USER_INPUT_PATH_ERR ) {
+                return make_error_response(status, "You tried to trim a path that doesn't seem to exist in this zone.");
+              }
+              else if ( status == USER_INCOMPATIBLE_PARAMS ) {
+                return make_error_response(status, "You have specified incompatible parameters for your trim operation.");
+              }
+	      else if ( status == CAT_NO_ROWS_FOUND) {
+                return make_error_response(status, "Could not find the logical path you entered. You may have passed an empty string.");
+              }
+              else if ( status < 0 ) {
+                return make_error_response(status, "Client experienced an error processing your trim request. Please check the logs or notify your administrator.");
+              }
+
+              return std::make_tuple(
+                Pistache::Http::Code::Ok,
+                ""
+              );
+            }
+            catch (const irods::exception& e) {
+                error("Caught exception - [error_code={}] {}", e.code(), e.what());
+                return make_error_response(e.code(), e.what());
+            }
+            catch (const std::exception& e) {
+                error("Caught exception - {}", e.what());
+                return make_error_response(SYS_INVALID_INPUT_PARAM, e.what());
+            }
+	    catch (...){
+		error("Caught unknown exception.");
+		return make_error_response(SYS_UNKNOWN_ERROR, "Caught unknown exception");
+	    }
+        }
+
+        std::tuple<Pistache::Http::Code, std::string>
+        replicate_dispatcher(const Pistache::Rest::Request& _request,
+                   Pistache::Http::ResponseWriter& _response)
+        {
+            try {
+              dataObjInp_t repl_input{};
```

Comment:
I suspect that the `condInput` member of `repl_input` may be leaking. We should set up an `at_scope_exit` to call `clearKeyVal` on it.

Reply:
yes please

Reply:
Done